index.js

```diff
@@ -64,8 +64,11 @@ async function fastifyRoutesStats (fastify, opts) {
     performance.mark(ONSEND + id)
 
     const key = `${ROUTES}${request.raw.method}|${routeId}`
-    performance.measure(key, ONREQUEST + id, ONSEND + id)
-
+    try {
```

Comment:
Cant we avoid the try catch?

Reply:
You should find the id not in the raw object but directly on request.
See
https://github.com/fastify/fastify/blob/4c416774ae20c5ea80a5301f0485877512a36da1/lib/request.js#L29

Reply:
also, can we do something like (adding the reply.mark param): 
``` 
 fastify.addHook('onRequest', function (request, reply, next) {
    const id = request.id
    performance.mark(ONREQUEST + id)
    reply.mark = true;
    next()
  })
```

And check the `mark` param in the `onSend` ? This way we don't need the try..catch,.. part

Reply:
@Uzlopak any feedback on the proposal ?

Reply:
I would rather analyze why the id was used in the first place. When using request.id then the unit tests fail. So something is odd.

Reply:
they don't seem to fail when i checked...